nftest/NFTestAssert.py

```diff
@@ -49,8 +51,36 @@ def get_assert_method(self) -> Callable:
             def func(actual, expect):
                 cmd = f"{self.script} {actual} {expect}"
                 self._logger.debug(cmd)
-                process_out = sp.run(cmd, shell=True, check=False)
-                return process_out.returncode == 0
+                with sp.Popen(cmd,
+                              shell=True,
+                              stdout=PIPE,
+                              stderr=PIPE,
+                              universal_newlines=True) as process:
+                    # Route stdout to INFO and stderr to ERROR in real-time
+                    with selectors.DefaultSelector() as selector:
+                        selector.register(
+                            fileobj=process.stdout,
+                            events=selectors.EVENT_READ,
+                            data=DEBUG
+                        )
+                        selector.register(
+                            fileobj=process.stderr,
+                            events=selectors.EVENT_READ,
+                            data=ERROR
+                        )
+
+                        while process.poll() is None:
+                            events = selector.select()
+                            for key, _ in events:
+                                line = key.fileobj.readline()
+                                if line:
+                                    # The only case in which this won't be true is when
+                                    # the pipe is closed
+                                    self._logger.log(
```

Comment:
It might be worthwhile to use a different logger so that these lines are easier to pick out, but that'd be a nicety.

Reply:
Makes sense, I've created an issue to track this for now